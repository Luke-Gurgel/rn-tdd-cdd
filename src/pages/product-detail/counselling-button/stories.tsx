import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { PageContainer } from '__storybook__/decorators/containers'
import CounsellingButton from '.'

storiesOf('Helpful button', module)
  .addDecorator(PageContainer)
  .add('affordable', () => <CounsellingButton price={20} canAfford />)
  .add('not affordable', () => (
    <CounsellingButton price={45} canAfford={false} />
  ))
