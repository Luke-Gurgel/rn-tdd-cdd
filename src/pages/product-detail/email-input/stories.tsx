import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { PageContainer } from '__storybook__/decorators/containers'
import EmailInput from '.'

storiesOf('Email input', module)
  .addDecorator(PageContainer)
  .add('default appearance', () => <EmailInput />)
  .add('email is required error', () => (
    <EmailInput error={{ type: 'required', message: 'Email is required' }} />
  ))
  .add('invalid email error', () => (
    <EmailInput
      error={{ type: 'invalid', message: 'Please enter a valid email' }}
    />
  ))
