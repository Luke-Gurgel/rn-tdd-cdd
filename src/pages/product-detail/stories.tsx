import React from 'react'
import { storiesOf } from '@storybook/react-native'
import ProductDetail from '.'

storiesOf('Product detail component', module).add('default appearance', () => (
  <ProductDetail />
))
