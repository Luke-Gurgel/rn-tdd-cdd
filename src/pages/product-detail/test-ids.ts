export const productDetail = 'product-detail'
export const emailInput = 'product-detail/email-input'
export const errorLabel = 'product-detail/error-label'
export const counsellingButton = 'product-detail/counselling-button'
