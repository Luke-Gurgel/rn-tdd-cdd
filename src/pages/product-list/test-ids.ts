export const productsList = 'products-list'
export const product = 'product-list/product/'
export const productsEmpty = 'products-list/empty'
export const productsLoading = 'products-list/loading'
