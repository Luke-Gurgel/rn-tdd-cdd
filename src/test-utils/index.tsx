import React from 'react'
import { render, RenderOptions } from '@testing-library/react-native'
import { Provider } from 'react-redux'
import store from 'src/store'
import { NavigationProvider } from 'react-navigation'

const AllProviders = (props: React.PropsWithChildren<any>) => {
  return (
    <Provider store={store}>
      <NavigationProvider
        value={{ dispatch: () => true, state: {}, navigate: () => {} }}>
        {props.children}
      </NavigationProvider>
    </Provider>
  )
}

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options })

export * from '@testing-library/react-native'

// override render method
export { customRender as render }
