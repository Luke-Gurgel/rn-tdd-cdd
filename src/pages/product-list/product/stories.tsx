import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { PageContainer } from '__storybook__/decorators/containers'
import {
  StoreProvider,
  NavigationProvider,
} from '__storybook__/decorators/providers'
import { ProductInterface } from 'src/types'
import Product from './'

export const products: { [key: string]: ProductInterface } = {
  aboveBudget: { title: 'Shoes', price: 100 },
  underBudget: { title: 'Shoes', price: 10 },
  equalBudget: { title: 'Shoes', price: 20 },
  longTitle: { title: 'Electric scooter by Movio', price: 53 },
}

storiesOf('Product component', module)
  .addDecorator(StoreProvider)
  .addDecorator(NavigationProvider)
  .addDecorator(PageContainer)
  .add('above budget', () => <Product product={products.aboveBudget} />)
  .add('under budget', () => <Product product={products.underBudget} />)
  .add('equal to budget', () => <Product product={products.equalBudget} />)
  .add('long product title', () => <Product product={products.longTitle} />)
