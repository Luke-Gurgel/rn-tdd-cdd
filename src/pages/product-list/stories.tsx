import React from 'react'
import { storiesOf } from '@storybook/react-native'
import ProductListUI from './index.ui'
import { ProductInterface } from 'src/types'
import {
  StoreProvider,
  NavigationProvider,
} from '__storybook__/decorators/providers'

export const products: ProductInterface[] = [
  { title: 'Shoes', price: 100 },
  { title: 'Shoes', price: 10 },
  { title: 'Shoes', price: 20 },
  { title: 'Electric scooter by Movio', price: 53 },
]

storiesOf('Product List component', module)
  .addDecorator(StoreProvider)
  .addDecorator(NavigationProvider)
  .add('fetching products', () => <ProductListUI loading products={[]} />)
  .add('empty list', () => <ProductListUI products={[]} loading={false} />)
  .add('product list with items', () => (
    <ProductListUI products={products} loading={false} />
  ))
