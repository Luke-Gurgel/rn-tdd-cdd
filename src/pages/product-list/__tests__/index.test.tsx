import React from 'react'
import 'react-native'
import ProductListUI from '../index.ui'
import { productsLoading, productsEmpty } from '../test-ids'
import { render } from 'src/test-utils'
import { products } from '../stories'

describe('ProductListUI component', () => {
  it('renders a loading component while the products are being fetched', () => {
    const { getByTestId } = render(<ProductListUI loading products={[]} />)
    expect(getByTestId(productsLoading)).toBeDefined()
  })

  it('renders a specific component when the product list is empty', () => {
    const { getByTestId } = render(
      <ProductListUI products={[]} loading={false} />,
    )
    expect(getByTestId(productsEmpty)).toBeDefined()
  })

  it('renders a grid of products if product list is not empty', () => {
    const { asJSON } = render(
      <ProductListUI products={products} loading={false} />,
    )
    expect(asJSON()).toMatchSnapshot()
  })
})
