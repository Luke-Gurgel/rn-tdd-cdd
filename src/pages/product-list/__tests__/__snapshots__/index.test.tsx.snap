// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListUI component renders a grid of products if product list is not empty 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "flex": 1,
        "paddingHorizontal": 20,
      }
    }
  >
    <ScrollView
      data={
        Array [
          Array [
            Object {
              "price": 100,
              "title": "Shoes",
            },
            Object {
              "price": 10,
              "title": "Shoes",
            },
            Object {
              "price": 20,
              "title": "Shoes",
            },
            Object {
              "price": 53,
              "title": "Electric scooter by Movio",
            },
          ],
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "paddingTop": 10,
          },
          Object {
            "flex": 1,
            "marginTop": 20,
          },
        ]
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 10,
              },
              Object {
                "marginBottom": 10,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "marginRight": 10,
                  "width": 138,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <TouchableOpacity
              activeOpacity={0.7}
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#FFF",
                  "borderRadius": 6,
                  "justifyContent": "center",
                  "paddingHorizontal": 15,
                  "paddingVertical": 15,
                  "width": 140,
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
              >
                Shoes
              </Text>
              <Text
                style={
                  Object {
                    "marginVertical": 4,
                  }
                }
              >
                $100.00
              </Text>
              <Text
                style={
                  Object {
                    "color": "#666",
                    "fontStyle": "italic",
                  }
                }
              >
                too expensive
              </Text>
            </TouchableOpacity>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "marginRight": 10,
                  "width": 138,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <TouchableOpacity
              activeOpacity={0.7}
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#FFF",
                  "borderRadius": 6,
                  "justifyContent": "center",
                  "paddingHorizontal": 15,
                  "paddingVertical": 15,
                  "width": 140,
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
              >
                Shoes
              </Text>
              <Text
                style={
                  Object {
                    "marginVertical": 4,
                  }
                }
              >
                $10.00
              </Text>
              <Text
                style={
                  Object {
                    "color": "#666",
                    "fontStyle": "italic",
                  }
                }
              >
                within reach!
              </Text>
            </TouchableOpacity>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "marginRight": 10,
                  "width": 138,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <TouchableOpacity
              activeOpacity={0.7}
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#FFF",
                  "borderRadius": 6,
                  "justifyContent": "center",
                  "paddingHorizontal": 15,
                  "paddingVertical": 15,
                  "width": 140,
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
              >
                Shoes
              </Text>
              <Text
                style={
                  Object {
                    "marginVertical": 4,
                  }
                }
              >
                $20.00
              </Text>
              <Text
                style={
                  Object {
                    "color": "#666",
                    "fontStyle": "italic",
                  }
                }
              >
                within reach!
              </Text>
            </TouchableOpacity>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "marginRight": 10,
                  "width": 138,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <TouchableOpacity
              activeOpacity={0.7}
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#FFF",
                  "borderRadius": 6,
                  "justifyContent": "center",
                  "paddingHorizontal": 15,
                  "paddingVertical": 15,
                  "width": 140,
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
              >
                Electric scooter by Movio
              </Text>
              <Text
                style={
                  Object {
                    "marginVertical": 4,
                  }
                }
              >
                $53.00
              </Text>
              <Text
                style={
                  Object {
                    "color": "#666",
                    "fontStyle": "italic",
                  }
                }
              >
                too expensive
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    </ScrollView>
  </View>
</View>
`;
